core-plugins/src/main/java/co/cask/hydrator/plugin/common/AvroToStructuredTransformer.java

```diff
@@ -71,4 +74,26 @@ public StructuredRecord transform(StructuredRecord.Builder builder, GenericRecor
 
     return builder.build();
   }
+
+  public StructuredRecord transform(GenericRecord genericRecord, String addFieldKey, String addFieldValue)
+    throws IOException {
+    Schema schema = Schema.parseJson(genericRecord.getSchema().toString());
+    Schema.Field newField = Schema.Field.of(addFieldKey, Schema.of(Schema.Type.STRING));
+    List<Schema.Field> newFields = new ArrayList<>();
+
+    for (Schema.Field field : schema.getFields()) {
+      newFields.add(field);
+    }
+    newFields.add(newField);
+    Schema newSchema = Schema.recordOf(schema.getRecordName(), newFields);
+
+    GenericRecordBuilder genericRecordBuilder = new GenericRecordBuilder(new org.apache.avro.Schema.Parser().
```

Comment:
don't build a brand new record and convert it, you'll double the work. You should be converting the GenericRecord once, and setting the pathfield at the very end on the StructuredRecord.Builder.